src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -58,10 +58,18 @@ public function __toString()
     /**
      * Returns the headers.
      *
+     * @param string|null $key The name of the headers to return or null to get them all
+     *
      * @return array An array of headers
      */
-    public function all()
+    public function all(/*string $key = null*/)
     {
+        if (1 <= \func_num_args() && null !== $key = func_get_arg(0)) {
+            $key = str_replace('_', '-', strtolower($key));
+
+            return $this->headers[$key] ?? [];
```

Comment:
This doesn't solve the problem in the linked issue (the `get` method having 2 different return types depending on a parameter), now it's `all` that can return either `string[]` or `string[][]` based on the presence of a parameter.

Reply:
that's fine to me, the proposed interface is a nice update